clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -988,4 +1056,27 @@ public int compare(ConfigKey k1, ConfigKey k2) {
         });
         return configs;
     }
+
+    /**
+     * Get a list of configs sorted into "group" order: listing group, then
+     * ordering by order in group.
+     */
+    private List<ConfigKey> sortedConfigsInGroup() {
+        // sort first with group, then by name
+        List<ConfigKey> configs = new ArrayList<>(this.configKeys.values());
+        Collections.sort(configs, new Comparator<ConfigKey>() {
+            public int compare(ConfigKey k1, ConfigKey k2) {
+                // sort by group
+                int cmp = k1.group.compareTo(k2.group);
```

Comment:
Need to check if `group` is `null` in both `k1` and `k2`. Using this on, e.g., `DistributedConfig` from Kafka Connect doesn't currently work.